core-bundle/contao/forms/FormText.php

```diff
@@ -245,6 +251,18 @@ protected function validator($varInput)
 		{
 			$varInput = Idna::encodeEmail($varInput);
 		}
+		elseif ($this->rgxp == 'date')
+		{
+			$targetFormat = Date::getNumericDateFormat();
+
+			// Check if date format matches the HTML5 standard
+			if (self::HTML5_DATE_FORMAT !== $targetFormat && preg_match('~^' . Date::getRegexp(self::HTML5_DATE_FORMAT) . '$~i', $varInput ?? ''))
```

Comment:
Why the `preg_match()` check?

Reply:
`\DateTimeImmutable::createFromFormat` will fail, if the input format does not match the expected format. Hence the check so that the date format is only transformed, if the input is actually valid.

Reply:
What happens if the input is invalid?

Reply:
Then `Widget::validator` will add an error (as usual).